crates/sui-core/src/authority_server.rs

```diff
@@ -909,6 +942,98 @@ impl ValidatorService {
         Ok((tonic::Response::new(response), Weight::one()))
     }
 
+    fn get_client_ip_addr<T>(&self, request: &tonic::Request<T>) -> Option<IpAddr> {
+        if self.client_id_source.is_none() {
+            return None;
+        }
```

Comment:
You may consider defaulting to the behavior of the `ClientIdSource::SocketAddr` variant case shown below (line 952-971) here, as perhaps something is better than nothing (which will be the case 95% of the time as most nodes will not have the necessary node config for `client_id_source` to be set. The reason the code you're referencing does an early return if `client_id_source` is not set is because the code needs to know that the IP it is collecting is indeed the IP of the originating client (or fullnode, in the validator case) and not some intermediate proxy. That is, nothing is better than something incorrect in that case. Feel free to use your best judgement here given this context

Reply:
Thanks for the input!

I guess one possible way is to add an argument to `get_client_ip_addr` to control this behavior (whether defaulting to the said variant), keep reusing this method, so that this will work for now.  Does that make sense?

Reply:
That makes perfect sense.  I have updated the PR according to this.